transport_analysis/analysis/velocityautocorr.py

```diff
@@ -5,111 +5,200 @@
+        Number of frames analysed in the trajectory.
+    n_particles : int
+        Number of particles VACF was calculated over.
     """
 
     def __init__(
         self,
-        universe_or_atomgroup: Union["Universe", "AtomGroup"],
-        select: str = "all",
-        # TODO: add your own parameters here
-        **kwargs
-    ):
+        atomgroup: "AtomGroup",
+        dim_type: str = "xyz",
+        fft: bool = True,
+        **kwargs,
+    ) -> None:
         # the below line must be kept to initialize the AnalysisBase class!
-        super().__init__(universe_or_atomgroup.trajectory, **kwargs)
+        super().__init__(atomgroup.universe.trajectory, **kwargs)
         # after this you will be able to access `self.results`
         # `self.results` is a dictionary-like object
         # that can should used to store and retrieve results
         # See more at the MDAnalysis documentation:
         # https://docs.mdanalysis.org/stable/documentation_pages/analysis/base.html?highlight=results#MDAnalysis.analysis.base.Results
 
-        self.universe = universe_or_atomgroup.universe
-        self.atomgroup = universe_or_atomgroup.select_atoms(select)
+        if isinstance(atomgroup, UpdatingAtomGroup):
+            raise TypeError(
+                "UpdatingAtomGroups are not valid for VACF " "computation"
+            )
+
+        # args
+        self.dim_type = dim_type
+        self._parse_dim_type()
+        self.fft = fft
+
+        # local
+        self.atomgroup = atomgroup
+        self.n_particles = len(self.atomgroup)
+        self._velocity_array = None
+
+        # result
+        self.results.vacf_by_particle = None
+        self.results.timeseries = None
 
     def _prepare(self):
-        """Set things up before the analysis loop begins"""
-        # This is an optional method that runs before
-        # _single_frame loops over the trajectory.
-        # It is useful for setting up results arrays
-        # For example, below we create an array to store
-        # the number of atoms with negative coordinates
-        # in each frame.
-        self.results.is_negative = np.zeros(
-            (self.n_frames, self.atomgroup.n_atoms),
-            dtype=bool,
+        """Set up velocity and VACF arrays before the analysis loop begins"""
+        # 2D array of frames x particles
+        self.results.vacf_by_particle = np.zeros(
+            (self.n_frames, self.n_particles)
+        )
+
+        # 3D array of frames x particles x dimensionality
+        self._velocity_array = np.zeros(
+            (self.n_frames, self.n_particles, self.dim_fac)
         )
+        # self.results.timeseries not set here
+
+    def _parse_dim_type(self):
+        r"""Sets up the desired dimensionality of the VACF."""
+        keys = {
+            "x": [0],
+            "y": [1],
+            "z": [2],
+            "xy": [0, 1],
+            "xz": [0, 2],
+            "yz": [1, 2],
+            "xyz": [0, 1, 2],
+        }
```

Comment:
It's mentioned above to include any ordering of `xyz`, which I think would be nice. A good way to implement that would be to convert the `dim_type` string to a set and then use sets as the keys here. That would give you flexibility without enumerating every option. e.g.

```python
dim_type = set(self.dim_type.split())

        keys = {
            {"x"}: [0],
            {"y"}: [1],
            {"z"}: [2],
            {"x", "y"}: [0, 1],
            {"x", "z"}: [0, 2],
            {"y", "z"}: [1, 2],
            {"x", "y", "z"}: [0, 1, 2],
        }

self._dim = keys[dim_type]

```

Reply:
This looks great! Would you be all right with me making this change in a separate PR? I'd like to get this one merged first to fix the CI (the check via Black is failing atm).

Reply:
Yep no problem. Maybe raise an issue and resolve this comment?

Reply:
Done in #13